docs/proposals/emptydir-pv.md

```diff
@@ -0,0 +1,151 @@
+# Emptydir persistent volume
+
+This document proposes a model for managing emptydir persistent volumes backed by node local storage.
+
+## Abstract
+We currently have:
+* Various different types of "local volumes", like emptyDir, hostPath and gitRepo
+* A way to "claim" storage, such that the pod referencing the claim can continue using the storage across pod/node restarts
+* A way to dynamically provision volumes to match un-bound/pending claims
+
+The second and third features allow us to say: provision a volume on demand, that survives pod/node restarts by using network attached storage (NFS, network block device from cloudprovider, distributed file system like gluster). This proposal is about a new `emptyDir` persistent volume that combines all three features. The idea is that we can sacrifice some durability to get a similar abstraction using *node local storage*. We expect this new PV source to be useful to applications that:
+* Prioritize latency over durability (eg: async replicated database using local ssd)
+* Prioritize cost over durability (eg: dev/test prototyping)
+* Bring their own out of band save/restore snapshotting
+
+## Goals
+* Define a new persistent volume source that is backed by local storage and survives pod restarts
+* Every pod referencing this PV must land on the same node (this concept has been referred to as "data gravity").
+* Once this PV is bound to a node, it cannot get unbound till a human intervenes or the node is deleted
+* The PV must require low user interaction. It should be provisionable dynamically.
+* The directory backing the PV must get deleted when the PV is deleted.
+
+## Non goals
+* A local storage abstraction that survives extended node outages
+* Snapshotting or simply copying the local volume to another node
+
+## A new PV storage source
+
+As described above, the storage source for the PV must be a local volume. While we already allow hostPath as a PersistentVolumeSource, it is insufficient because:
+* user needs to specify the exact path, thus requires knowledge of the filesystem
```

Comment:
This is the idea proposed in the alternatives. The emptyDir approach is neater IMO because it gives you writeable space *somewhere* on the node even without the dynamic provisioner, much like emptyDir already works. The user doesn't have to care about anything else.